api/client/admin/storagejobs.go

```diff
@@ -4,96 +4,98 @@ import (
 	"context"
 
 	adminPb "github.com/textileio/powergate/api/gen/powergate/admin/v1"
+	userPb "github.com/textileio/powergate/api/gen/powergate/user/v1"
 )
 
 // StorageJobs provides access to Powergate jobs admin APIs.
 type StorageJobs struct {
 	client adminPb.AdminServiceClient
 }
 
-type storageJobsConfig struct {
-	UserID string
-	Cids   []string
-}
+// ListSelect specifies which StorageJobs to list.
+type ListSelect int32
 
-// StorageJobsOption configures a storageJobsConfig.
-type StorageJobsOption = func(*storageJobsConfig)
+const (
+	// All lists all StorageJobs and is the default.
+	All ListSelect = iota
+	// Queued lists queued StorageJobs.
+	Queued
+	// Executing lists executing StorageJobs.
+	Executing
+	// Final lists final StorageJobs.
+	Final
+)
 
-// WithUserID filters the results to the specified user.
-func WithUserID(userID string) StorageJobsOption {
-	return func(conf *storageJobsConfig) {
-		conf.UserID = userID
-	}
+// ListConfig controls the behavior for listing StorageJobs.
+type ListConfig struct {
+	// UserIDFilter filters StorageJobs list to the specified user ID. Defaults to no filter.
+	UserIDFilter string
+	// CidFilter filters StorageJobs list to the specified cid. Defaults to no filter.
+	CidFilter string
+	// Limit limits the number of StorageJobs returned. Defaults to no limit.
+	Limit uint64
```

Comment:
Yes, I can set some default limit for remote clients. I think on the server side, the default should still be no limit so that in-process powergate calls can just get whatever data they want fast without having to page.

Reply:
SGTM.

Reply:
I'm going to hold of on adding a default limit to the client until a later pr where we actually add support for processing paged data to the cli. Don't feel like biting that off right now.

Reply:
https://github.com/textileio/powergate/issues/749